core/src/Streamly/Internal/Data/Stream/PreludeCommon.hs

```diff
@@ -0,0 +1,139 @@
+------------------------------------------------------------------------------
```

Comment:
Raise an issue to remove this module and the Data.Stream.Prelude module, and assimilate these into Serial module. Target the issue to 0.9.0.

We can also assimilate Instances.hs in the serial module, only those instances that are common with the Zip module or any other module in this package can be kept in the common CPP header.